File: amazon/API/amzn/Orders/generateExcelFiles.rb
#!/usr/local/rvm/rubies/default/bin/ruby
#require 'rubygems'
require 'open-uri'
require 'csv'
require 'pp'
require 'axlsx'
url = '/home/seanno/public_html/admin/amazon-unshipped-orders.csv'
#url = 'http://www.lifehousemedical.com/admin/amazon-unshipped-orders.csv'
datestamp = Time.now.strftime('%Y%m%d')
original_csv = []
basedir='/home/seanno/public_html/admin'

amz_array = []
ovation_array = []
patterson_array = []
isg_array = []
medline_array = []
indemed_array = []
nbs_array = []

all_orderids_array = []
ovation_orderids_array = []
patterson_orderids_array = []
isg_orderids_array = []
medline_orderids_array = []
indemed_orderids_array = []

#puts "Fetching #{url}..."
#puts ''

#puts file
#CSV.foreach(url, :headers => :first_row).each do |line|
#CSV.foreach(url) do |line|

CSV.new(open(url), :headers => :first_row).each do |line|
if line['ship-postal-code'] && line['ship-postal-code'] != ''
	line['ship-postal-code'].to_s! if line['ship-postal-code'].class != String
end
	original_csv << line.to_hash
	amz_array << line.to_hash
end

Axlsx::Package.new do |p|
	p.workbook.add_worksheet(:name => 'AMZ') do |sheet|
		text_format = p.workbook.styles.add_style :format_code => '@'
		yellow_format = p.workbook.styles.add_style :format_code => '@', :bg_color => 'FFFF00'
		blue_format = p.workbook.styles.add_style :format_code => '@', :bg_color => '00FFFF'
		sheet.add_row ['Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 
			'Quantity', 'ShipService', 'PurchaseOrder']
		original_csv.each do |row|
			formatting_array = [text_format] * 11
			if (row['quantity-to-ship'].to_i > 1)
				formatting_array[8] = yellow_format
			end
			if (row['ship-service-level'] != "Standard")
				formatting_array[9] = blue_format
			end 
			sheet.add_row [row['recipient-name'], row['ship-address-1'], row['ship-address-2'], 
				row['ship-city'], row['ship-state'], "=\"#{row['ship-postal-code']}\"", row['sku'], 
				row['product-name'], row['quantity-to-ship'], row['ship-service-level'], "=\"#{row['order-id'].gsub("'","")}\""], :style => formatting_array
		end
		sheet.column_widths nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil
	end
	p.serialize(File.expand_path(basedir + "/amazon-templates/AMZ-#{datestamp}.xlsx"))
end

original_csv.each do |row|
	all_orderids_array << row['order-id'];
	if (row['sku'].include?('NMI')) && (!row['product-name'].include?('Ovation')) && (!row['product-name'].include?('Adjustable Plantar Fasciitis'))
		nbs_array << row
		next
	end
	if (row['product-name'].include?('Ovation')) || (row['product-name'].include?('Adjustable Plantar Fasciitis'))
		ovation_array << row
		ovation_orderids_array << row['order-id'];
		next
	end
	if (row['sku'].include?('CTS'))
		patterson_array << row
		patterson_orderids_array << row['order-id'];
		next
	end
	if (row['sku'].include?('MD44')) || (row['sku'].include?('MDLN')) || (row['sku'].include?('MD88'))
		medline_array << row
		medline_orderids_array << row['order-id'];
		next
	end
	if ((row['sku'].count('-') == 0) && (!row['product-name'].include?('Ovation'))) || (((row['sku'].include?('IND')) || (row['sku'].include?('UHS')) && (row['sku'].count('-') == 1)))
		indemed_array << row
		indemed_orderids_array << row['order-id'];
		next
	end
	if (row['sku'].count('-') > 1)
		isg_array << row
		isg_orderids_array << row['order-id'];
		next
	end
	nbs_array << row
end

Axlsx::Package.new do |p|
	p.workbook.add_worksheet(:name => 'NBS') do |sheet|
		text_format = p.workbook.styles.add_style :format_code => '@'
		yellow_format = p.workbook.styles.add_style :format_code => '@', :bg_color => 'FFFF00'
		blue_format = p.workbook.styles.add_style :format_code => '@', :bg_color => '00FFFF'
		sheet.add_row ['Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 
			'Quantity', 'ShipService', 'PurchaseOrder']
		ovation_array.each do |row|
			formatting_array = [text_format] * 11
			if (row['quantity-to-ship'].to_i > 1)
				formatting_array[8] = yellow_format
			end
			if (row['ship-service-level'] != "Standard")
				formatting_array[9] = blue_format
			end 
			sheet.add_row [row['recipient-name'], row['ship-address-1'], row['ship-address-2'], 
				row['ship-city'], row['ship-state'], "=\"#{row['ship-postal-code']}\"", row['sku'], 
				row['product-name'], row['quantity-to-ship'], row['ship-service-level'], "=\"#{row['order-id'].gsub("'","")}\""], :style => formatting_array
		end
		sheet.column_widths nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil
	end
	p.serialize(File.expand_path(basedir + "/amazon-templates/Ovation-#{datestamp}.xlsx"))
end

Axlsx::Package.new do |p|
	p.workbook.add_worksheet(:name => 'NBS') do |sheet|
		text_format = p.workbook.styles.add_style :format_code => '@'
		yellow_format = p.workbook.styles.add_style :format_code => '@', :bg_color => 'FFFF00'
		blue_format = p.workbook.styles.add_style :format_code => '@', :bg_color => '00FFFF'
		sheet.add_row ['Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 
			'Quantity', 'ShipService', 'PurchaseOrder']
		nbs_array.each do |row|
			formatting_array = [text_format] * 11
			if (row['quantity-to-ship'].to_i > 1)
				formatting_array[8] = yellow_format
			end
			if (row['ship-service-level'] != "Standard")
				formatting_array[9] = blue_format
			end 
			sheet.add_row [row['recipient-name'], row['ship-address-1'], row['ship-address-2'], 
				row['ship-city'], row['ship-state'], "=\"#{row['ship-postal-code']}\"", row['sku'], 
				row['product-name'], row['quantity-to-ship'], row['ship-service-level'], "=\"#{row['order-id'].gsub("'","")}\""], :style => formatting_array
		end
		sheet.column_widths nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil
	end
	p.serialize(File.expand_path(basedir + "/amazon-templates/NBS-#{datestamp}.xlsx"))
end

Axlsx::Package.new do |p|
	p.workbook.add_worksheet(:name => 'Patterson') do |sheet|
		text_format = p.workbook.styles.add_style :format_code => '@'
		yellow_format = p.workbook.styles.add_style :format_code => '@', :bg_color => 'FFFF00'
		blue_format = p.workbook.styles.add_style :format_code => '@', :bg_color => '00FFFF'
		sheet.add_row ['Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 
			'Quantity', 'ShipService', 'PurchaseOrder']
		patterson_array.each do |row|
			formatting_array = [text_format] * 11
			if (row['quantity-to-ship'].to_i > 1)
				formatting_array[8] = yellow_format
			end
			if (row['ship-service-level'] != "Standard")
				formatting_array[9] = blue_format
			end 
			sheet.add_row [row['recipient-name'], row['ship-address-1'], row['ship-address-2'], 
				row['ship-city'], row['ship-state'], "=\"#{row['ship-postal-code']}\"", row['sku'].gsub("CTS101-",""), 
				row['product-name'], row['quantity-to-ship'], row['ship-service-level'], "=\"#{row['order-id'].gsub("'","")}\""], :style => formatting_array
		end
		sheet.column_widths nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil
	end
	p.serialize(File.expand_path(basedir + "/amazon-templates/Patterson-#{datestamp}.xlsx"))
end

Axlsx::Package.new do |p|
	p.workbook.add_worksheet(:name => 'Medline') do |sheet|
		text_format = p.workbook.styles.add_style :format_code => '@'
		yellow_format = p.workbook.styles.add_style :format_code => '@', :bg_color => 'FFFF00'
		blue_format = p.workbook.styles.add_style :format_code => '@', :bg_color => '00FFFF'
		sheet.add_row ['Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 
			'Quantity', 'ShipService', 'PurchaseOrder']
		medline_array.each do |row|
			formatting_array = [text_format] * 11
			if (row['quantity-to-ship'].to_i > 1)
				formatting_array[8] = yellow_format
			end
			if (row['ship-service-level'] != "Standard")
				formatting_array[9] = blue_format
			end 
			sheet.add_row [row['recipient-name'], row['ship-address-1'], row['ship-address-2'], 
				row['ship-city'], row['ship-state'], "=\"#{row['ship-postal-code']}\"", row['sku'].gsub(/MDLN1|MD44|MD88|MDLN-/,""), 
				row['product-name'], row['quantity-to-ship'], row['ship-service-level'], "=\"#{row['order-id'].gsub("'","")}\""], :style => formatting_array
		end
		sheet.column_widths nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil
	end
	p.serialize(File.expand_path(basedir + "/amazon-templates/Medline-#{datestamp}.xlsx"))
end

Axlsx::Package.new do |p|
	p.workbook.add_worksheet(:name => 'Indemed') do |sheet|
		text_format = p.workbook.styles.add_style :format_code => '@'
		yellow_format = p.workbook.styles.add_style :format_code => '@', :bg_color => 'FFFF00'
		blue_format = p.workbook.styles.add_style :format_code => '@', :bg_color => '00FFFF'
		sheet.add_row ['Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 
			'Quantity', 'ShipService', 'PurchaseOrder']
		indemed_array.each do |row|
			formatting_array = [text_format] * 11
			if (row['quantity-to-ship'].to_i > 1)
				formatting_array[8] = yellow_format
			end
			if (row['ship-service-level'] != "Standard")
				formatting_array[9] = blue_format
			end 
			sheet.add_row [row['recipient-name'], row['ship-address-1'], row['ship-address-2'], 
				row['ship-city'], row['ship-state'], "=\"#{row['ship-postal-code']}\"", row['sku'].gsub(/IND-|INDP-|UHS-/,""), 
				row['product-name'], row['quantity-to-ship'], row['ship-service-level'], "=\"#{row['order-id'].gsub("'","")}\""], :style => formatting_array
		end
		sheet.column_widths nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil
	end
	p.serialize(File.expand_path(basedir + "/amazon-templates/Indemed-#{datestamp}.xlsx"))
end

Axlsx::Package.new do |p|
	p.workbook.add_worksheet(:name => 'ISG') do |sheet|
		text_format = p.workbook.styles.add_style :format_code => '@'
		yellow_format = p.workbook.styles.add_style :format_code => '@', :bg_color => 'FFFF00'
		blue_format = p.workbook.styles.add_style :format_code => '@', :bg_color => '00FFFF'
		sheet.add_row ['Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 
			'Quantity', 'ShipService', 'PurchaseOrder']
		isg_array.each do |row|
			formatting_array = [text_format] * 11
			if (row['quantity-to-ship'].to_i > 1)
				formatting_array[8] = yellow_format
			end
			if (row['ship-service-level'] != "Standard")
				formatting_array[9] = blue_format
			end 
			sheet.add_row [row['recipient-name'], row['ship-address-1'], row['ship-address-2'], 
				row['ship-city'], row['ship-state'], "=\"#{row['ship-postal-code']}\"", row['sku'].gsub(/UHS-|ISG-/,""), 
				row['product-name'], row['quantity-to-ship'], row['ship-service-level'], "=\"#{row['order-id'].gsub("'","")}\""], :style => formatting_array
		end
		sheet.column_widths nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil
	end
	p.serialize(File.expand_path(basedir + "/amazon-templates/ISG-#{datestamp}.xlsx"))
end

order_ids = all_orderids_array.join(",").gsub("'", '')
f = File.new(basedir + '/amazon-templates/all.csv', 'w')
f.write("#{order_ids}\n")
f.close();

order_ids = ovation_orderids_array.join(",").gsub("'", '')
f = File.new(basedir + '/amazon-templates/ovation.csv', 'w')
f.write("#{order_ids}\n")
f.close();

order_ids = patterson_orderids_array.join(",").gsub("'", '')
f = File.new(basedir + '/amazon-templates/patterson.csv', 'w')
f.write("#{order_ids}\n")
f.close();

order_ids = isg_orderids_array.join(",").gsub("'", '')
f = File.new(basedir + '/amazon-templates/isg.csv', 'w')
f.write("#{order_ids}\n")
f.close();

order_ids = medline_orderids_array.join(",").gsub("'", '')
f = File.new(basedir + '/amazon-templates/medline.csv', 'w')
f.write("#{order_ids}\n")
f.close();

order_ids = indemed_orderids_array.join(",").gsub("'", '')
f = File.new(basedir + '/amazon-templates/indemed.csv', 'w')
f.write("#{order_ids}\n")
f.close();

#puts "Original CSV:\t\t#{original_csv.count} record(s)."
#puts "AMZ-#{datestamp}:\t\t#{amz_array.count} record(s)"
#puts "Patterson-#{datestamp}:\t#{patterson_array.count} record(s)"
#puts "Indemed-#{datestamp}:\t#{indemed_array.count} record(s)"
#puts "ISG-#{datestamp}:\t\t#{isg_array.count} record(s)"
#puts "NBS-#{datestamp}:\t\t#{nbs_array.count} record(s)"
#puts "Medline-#{datestamp}:\t#{medline_array.count} record(s)"
#
#puts "Total:\t\t\t#{patterson_array.count + indemed_array.count + isg_array.count + nbs_array.count + medline_array.count} record(s)"
#puts ''
#puts 'Excel files successfully generated. You may now close this window.'

exit

File: amazon/API/amzn/Orders/generateExcelFiles.php
<?php
$BASEDIR = $_SERVER['DOCUMENT_ROOT'].'/'.FOLDER_NAME.'';

# include parseCSV class.
require_once($_SERVER['DOCUMENT_ROOT'] .'/projects/inventory_api/parsecsv/parsecsv.lib.php');

$url = $BASEDIR.'/amazon-unshipped-orders.csv';
#url = 'http://tops-tech.com/amzn_api/amazon-unshipped-orders.csv'
$datestamp = strftime('%Y%m%d');
$original_csv = array();
//$BASEDIR='/home7/topstech/public_html/amzn_api';

$amz_array = array();
$ovation_array = array();
$patterson_array = array();
$isg_array = array();
$medline_array = array();
$indemed_array = array();
$nbs_array = array();

$all_orderids_array = array();
$ovation_orderids_array = array();
$patterson_orderids_array = array();
$isg_orderids_array = array();
$medline_orderids_array = array();
$indemed_orderids_array = array();

#puts "Fetching #{url}..."
#puts ''
$new_csv=array();
#puts file
#CSV.foreach(url, :headers => :first_row).each do |line|
#CSV.foreach(url) do |line|
$csv = new parseCSV();
# Parse '_books.csv' using automatic delimiter detection...
$csv->auto($url);

/*$i=0;
foreach ($csv->data as $key => $row){
	
	//foreach ($row as $value){
		
		$new_csv[$i] =  $csv->data;
		
	//}
$i+$i+1;	
}*/

echo "<pre>";
	$i=0;
	
	$AMZ_FILE_PATH = 'amazon-templates/AMZ-'.$datestamp.'.csv';
	
	$fp = fopen($AMZ_FILE_PATH , 'w');
	
	$AMZ_TITLE = array('Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 'Quantity', 'ShipService', 'PurchaseOrder');
	//foreach ($AMZ_TITLE as $title){
		fputcsv($fp,$AMZ_TITLE);
	//}
	
	//foreach ($csv->titles as $value){
		//$AMZ_TITLE[$i] =  $csv->titles;
		//fputcsv($fp,$AMZ_TITLE[$i]);
		//$i++;
	//}
	//print_r($AMZ_TITLE);exit;
	
	
	
	$i=0;
	
	foreach ($csv->data as $key => $row){
		//foreach ($row as $value){
			array_push($original_csv,$row);
			//$AMZ_TITLE[$i] =  $row;
		//}
		
		/*if($row['order-id']){ 
			echo $row['order-id'].' => ';
			$row['order-id'] = str_replace("'","",strval($row['order-id'])); 
			echo $row['order-id'].'\n';
		}*/
		
		fputcsv($fp, $row);

		// Insert Values into Arrays
		#$string = preg_replace("/[']/", '', $row['order-id']);
		$string = $row['order-id'];
		
		array_push($all_orderids_array,$string);
		#array_push($all_orderids_array,str_replace("'","",$string));
		
		if (strpos($row['sku'],'NMI')>0 && !strpos($row['product-name'],'Ovation') && !strpos($row['product-name'],'Adjustable Plantar Fasciitis') ){
			//echo strpos($row['sku'],'NMI');exit;
			array_push($nbs_array,$row);
			//$nbs_array[$i] = $row;
			continue;
		}
		//else
		if (strpos($row['product-name'],'Ovation')>0 || strpos($row['product-name'],'Adjustable Plantar Fasciitis')>0 ){
			array_push($ovation_array,$row);
			array_push($ovation_orderids_array,$string);
			
			/*$ovation_array[$i] = $row;
			$ovation_orderids_array[$i] = $string;*/
			continue;
		}
		//else
		if (strpos($row['sku'],'CTS')>0){
			array_push($patterson_array,$row);
			array_push($patterson_orderids_array,$string);
			
			/*$patterson_array[$i] = $row;
			$patterson_orderids_array[$i] = $string;*/
			continue;
		}
		//else
		if (strpos($row['sku'],'MD44')>0 || strpos($row['sku'],'MDLN')>0 || strpos($row['sku'],'MD88')>0 ){
			array_push($medline_array,$row);
			array_push($medline_orderids_array,$string);
			
			/*$medline_array = $row;
			$medline_orderids_array =  $string;*/
			continue;
		}
		//else
		if ( (substr_count($row['sku'],'-') == 0 && !strpos($row['product-name'],'Ovation')) || strpos($row['sku'],'IND')>0 || (strpos($row['sku'],'UHS')>0 && substr_count($row['sku'],'-') == 1) ){
			//echo strpos($row['product-name'],'Ovation');exit;
			array_push($indemed_array,$row);
			array_push($indemed_orderids_array,$string);
			
			/*$indemed_array[$i] = $row;
			$indemed_orderids_array[$i] = $string;*/
			continue;
		}
		//else
		if (substr_count($row['sku'],'-') > 1){
			array_push($isg_array,$row);
			array_push($isg_orderids_array,$string);
			
			/*$isg_array[$i] = $row;
			$isg_orderids_array[$i] = $string;*/
			continue;
		}
		array_push($nbs_array,$row);
		
		
		//$nbs_array[$i] = $row;
		
	$i++;	
	} # End Froreach;
	fclose($fp);
	chmod($AMZ_FILE_PATH, 0777);
	
	
	//print_r($AMZ_ROW);exit;
	//print_r($AMZ_TITLE);exit;
	
	/*echo "<br><br>all_orderids_array=> ";
	print_r($all_orderids_array);
	
	echo "<br><br>nbs_array=> ";
	print_r($nbs_array);
	
	echo "<br><br>ovation_array=> ";
	print_r($ovation_array);
	echo "<br><br>ovation_orderids_array=> ";	
	print_r($ovation_orderids_array);

	echo "<br><br>patterson_array=> ";	
	print_r($patterson_array);
	echo "<br><br>patterson_orderids_array=> ";
	print_r($patterson_orderids_array);
	
	echo "<br><br>medline_array=> ";
	print_r($medline_array);
	echo "<br><br>medline_orderids_array=> ";	
	print_r($medline_orderids_array);
	
	echo "<br><br>indemed_array=> ";	
	print_r($indemed_array);
	echo "<br><br>indemed_orderids_array=> ";	
	print_r($indemed_orderids_array);
	
	echo "<br><br>isg_array=> ";
	print_r($isg_array);
	echo "<br><br>isg_orderids_array=> ";	
	print_r($isg_orderids_array);*/
	
	
	// Create Ovation CSV
	//if(count($ovation_array)>0){	
		$Ovation_FILE_PATH = 'amazon-templates/Ovation-'.$datestamp.'.csv';
		$fp_Ovation = fopen($Ovation_FILE_PATH , 'w');
		
		$Ovation_TITLE = array('Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 'Quantity', 'ShipService', 'PurchaseOrder');
	
			fputcsv($fp_Ovation,$Ovation_TITLE);
			
		$i=0;
		foreach ($ovation_array as $key => $row){
				fputcsv($fp_Ovation, $row);
		}
		fclose($fp_Ovation);
		chmod($Ovation_FILE_PATH, 0777);
	//}
	
	
	
	// Create NBS Array CSV
	//if(count($nbs_array)>0){	
		$NBS_FILE_PATH = 'amazon-templates/NBS-'.$datestamp.'.csv';
		$fp_NBS = fopen($NBS_FILE_PATH , 'w');
		
		$NBS_TITLE = array('Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 'Quantity', 'ShipService', 'PurchaseOrder');
	
			fputcsv($fp_NBS,$NBS_TITLE);
			
		$i=0;
		foreach ($nbs_array as $key => $row){
				fputcsv($fp_NBS, $row);
		}
		fclose($fp_NBS);
		chmod($NBS_FILE_PATH, 0777);
	//}
	
	
	// Create Patterson Array CSV
	//if(count($patterson_array)>0){	
		$Patterson_FILE_PATH = 'amazon-templates/Patterson-'.$datestamp.'.csv';
		$fp_Patterson = fopen($Patterson_FILE_PATH , 'w');
		
		$Patterson_TITLE = array('Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 'Quantity', 'ShipService', 'PurchaseOrder');
	
			fputcsv($fp_Patterson,$Patterson_TITLE);
			
		$i=0;
		foreach ($patterson_array as $key => $row){
				fputcsv($fp_Patterson, $row);
		}
		fclose($fp_Patterson);
		chmod($Patterson_FILE_PATH, 0777);
	//}
	
	// Create Medline Array CSV
	//if(count($medline_array)>0){	
		$Medline_FILE_PATH = 'amazon-templates/Medline-'.$datestamp.'.csv';
		$fp_Medline = fopen($Medline_FILE_PATH , 'w');
		
		$Medline_TITLE = array('Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 'Quantity', 'ShipService', 'PurchaseOrder');
	
			fputcsv($fp_Medline,$Medline_TITLE);
			
		$i=0;
		foreach ($medline_array as $key => $row){
				fputcsv($fp_Medline, $row);
		}
		fclose($fp_Medline);
		chmod($Medline_FILE_PATH, 0777);
	//}
	
	
	// Create Indemed Array CSV
	//if(count($indemed_array)>0){	
		$Indemed_FILE_PATH = 'amazon-templates/Indemed-'.$datestamp.'.csv';
		$fp_Indemed = fopen($Indemed_FILE_PATH , 'w');
		
		$Indemed_TITLE = array('Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 'Quantity', 'ShipService', 'PurchaseOrder');
	
			fputcsv($fp_Indemed,$Indemed_TITLE);
			
		$i=0;
		foreach ($indemed_array as $key => $row){
				fputcsv($fp_Indemed, $row);
		}
		fclose($fp_Indemed);
		chmod($Indemed_FILE_PATH, 0777);
	//}
	
	// Create NBS Array CSV
	//if(count($isg_array)>0){	
		$ISG_FILE_PATH = 'amazon-templates/ISG-'.$datestamp.'.csv';
		$fp_ISG = fopen($ISG_FILE_PATH , 'w');
		
		$ISG_TITLE = array('Name', 'Address1', 'Address2', 'City', 'State', 'ZIP', 'SKU', 'ProductName', 'Quantity', 'ShipService', 'PurchaseOrder');
	
			fputcsv($fp_ISG,$ISG_TITLE);
			
		$i=0;
		foreach ($isg_array as $key => $row){
				fputcsv($fp_ISG, $row);
		}
		fclose($fp_ISG);
		chmod($ISG_FILE_PATH, 0777);
	//}
	
	
	
	$order_ids=array();
	
	// NoW Create Orders CSV Files
	
	# all_orderids_array CSV
	$all_orderids_array_FILE_PATH = 'amazon-templates/all.csv';
	$fp_all_orderids_array = fopen($all_orderids_array_FILE_PATH , 'w');

		fputcsv($fp_all_orderids_array,$all_orderids_array);	
	
	fclose($fp_all_orderids_array);
	chmod($all_orderids_array_FILE_PATH, 0777);
	
	
	# ovation_orderids_array CSV
	$ovation_orderids_array_FILE_PATH = 'amazon-templates/ovation.csv';
	$fp_ovation_orderids_array = fopen($ovation_orderids_array_FILE_PATH , 'w');

		fputcsv($fp_ovation_orderids_array,$ovation_orderids_array);	
	
	fclose($fp_ovation_orderids_array);
	chmod($ovation_orderids_array_FILE_PATH, 0777);
	
	# patterson_orderids_array CSV
	$patterson_orderids_array_FILE_PATH = 'amazon-templates/patterson.csv';
	$fp_patterson_orderids_array = fopen($patterson_orderids_array_FILE_PATH , 'w');

		fputcsv($fp_patterson_orderids_array,$patterson_orderids_array);	
	
	fclose($fp_patterson_orderids_array);
	chmod($patterson_orderids_array_FILE_PATH, 0777);
	
	# isg_orderids_array CSV
	$isg_orderids_array_FILE_PATH = 'amazon-templates/isg.csv';
	$fp_isg_orderids_array = fopen($isg_orderids_array_FILE_PATH , 'w');

		fputcsv($fp_isg_orderids_array,$isg_orderids_array);	
	
	fclose($fp_isg_orderids_array);
	chmod($isg_orderids_array_FILE_PATH, 0777);
	
	
	# medline_orderids_array CSV
	$medline_orderids_array_FILE_PATH = 'amazon-templates/medline.csv';
	$fp_medline_orderids_array = fopen($medline_orderids_array_FILE_PATH , 'w');

		fputcsv($fp_medline_orderids_array,$medline_orderids_array);	
	
	fclose($fp_medline_orderids_array);
	chmod($medline_orderids_array_FILE_PATH, 0777);
	
	
	# indemed_orderids_array CSV
	$indemed_orderids_array_FILE_PATH = 'amazon-templates/indemed.csv';
	$fp_indemed_orderids_array = fopen($indemed_orderids_array_FILE_PATH , 'w');

		fputcsv($fp_indemed_orderids_array,$indemed_orderids_array);	
	
	fclose($fp_indemed_orderids_array);
	chmod($indemed_orderids_array_FILE_PATH, 0777);
	
	
	echo "<br>";
	echo "Original CSV:\t\t".count($original_csv)." record(s)<br>";
	echo "AMZ-".$datestamp.":\t\t".count($amz_array)." record(s)<br>";
	echo "Patterson-".$datestamp.":\t".count($patterson_array)." record(s)<br>";
	echo "Indemed-".$datestamp.":\t".count($indemed_array)." record(s)<br>";
	echo "ISG-".$datestamp.":\t\t".count($isg_array)." record(s)<br>";
	echo "NBS-".$datestamp.":\t\t".count($nbs_array)." record(s)<br>";
	echo "Medline-".$datestamp.":\t".count($medline_array)." record(s)<br>";
	echo "--------------------------------------<br>";
	echo "Total:\t\t\t".(count($patterson_array) + count($indemed_array) + count($isg_array) + count($nbs_array) + count($medline_array))." record(s)<br><br>";

	echo 'Excel files successfully generated. You may now close this window.<br>';

	
exit

?>